api/serializers.py

```diff
@@ -12,22 +12,46 @@
 class ProfileSerializer(serializers.ModelSerializer):
     def to_representation(self, instance):
         rep = super().to_representation(instance)
-        # get video
-        video = None
-        try:
-            video = Video.objects.get(
-                Q(sender=instance.user) & Q(receiver=instance.user)
-            )
-            video = read_video(video.video_file)
-        finally:
-            rep["video"] = video
-            return rep
+
+        videoData = None
+        videoInstance = getVideoInstance(instance.user, instance.user)
+        if videoInstance is not None:
+            videoData = read_video(videoInstance.video_file)
+            # TODO: if videoData comes back none, then video file is lost,
+            # should we delete the video instance?
+
+        rep["video"] = videoData
+        return rep
 
     class Meta:
         model = Profile
         exclude = ("user",)
 
 
+def getVideoInstance(sender, receiver):
+    videoInstance = None
+    allVideoInstances = []
+
+    try:
+        # Get video instance.
+        allVideoInstances = Video.objects.filter(sender=sender, receiver=receiver)
+    except Exception as e:
+        # Video instance does not exist.
+        print(e)
```

Comment:
yeah I don't think I did a great job explaining myself in my comments in the elif clause below. I think a `Video.objects.get` is what we really want here since there should probably only be one intro video at a time for any user. I guess it doesn't matter either way but a get makes more sense to me. the question is how do we want to enforce the single intro video restraint

Reply:
I'm not convinced this is inside the scope of your PR